clarity/contracts/arkadiko-alex-dual-yield-v1-1.clar

```diff
@@ -0,0 +1,75 @@
+;; @contract ALEX Dual Yield - Mints DIKO for ALEX dual yield efforts
+;; @version 1.1
+
+;; Errors
+(define-constant ERR-NOT-AUTHORIZED u32401)
+(define-constant ERR-NOT-ACTIVATED u320001)
+(define-constant ERR-TOO-MUCH-DIKO u320002)
+(define-constant MAX-TO-MINT u1250000000000) ;; 1.25M DIKO
+
+(define-data-var diko-minted uint u0)
+(define-data-var rewards-per-cycle uint u12500000000) ;; 12.5K DIKO
+(define-data-var shutdown-activated bool false)
+
+(define-read-only (is-activated)
+  (and
+    (is-eq (unwrap-panic (contract-call? .arkadiko-dao get-emergency-shutdown-activated)) false)
+    (is-eq (var-get shutdown-activated) false)
+  )
+)
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; public methods (callable by ALEX contract or Arkadiko DAO) ;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(define-public (mint (amount uint) (recipient principal))
```

Comment:
@philipdesmedt given what we just discussed, I see that all we need to do is to call at each cycle  `mint (amount uint) (recipient principal)` with `amount` <= 12.5k (with the expectation being the sum of `amount` across all recipients for a given cycle would equal to 12.5k - this expectation will be codified into our contract).

Reply:
@fiftyeightandeight correct - each cycle you can give away 12.5K DIKO (12.5K DIKO if there would be only 1 LP).

Each LP should get pro rata an amount of DIKO vs their LP tokens staked and the sum of these amounts per cycle is 12.5K DIKO

Reply:
@philipdesmedt for 12.5k DIKO, `amount` should be 12.5k * 1e6 (i.e. 6 decimals for DIKO), correct?

Reply:
@fiftyeightandeight yes, correct! Always pass it with 6 decimals